### PR TITLE
added list of successful http codes

### PR DESCRIPTION
**Pull request by hellaxe**
**Issue**
API response from nylas GET `/oauth/authorize` returns HTTP=302 with empty body that causes call raise of exception (due it is not 200 code), but fails to do it due the empty body.

**Solution**
I have added list of successful HTTP codes (add more if your API responds with another non-error codes) to the check in `#handle_anticipated_failure_mode` method.

I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by arunagw** [comment]
Hey @hellaxe thanks for the PR. This seems reasonable but I don't see 302 response code in the documentation mentioned here https://developer.nylas.com/docs/api/#get/oauth/authorize

Can you please give us an example how you are calling this so that I can verify this by running.

**Comment by hellaxe** [comment]
Hey @arunagw . Thanks for the feedback. Indeed there are no 302 status code in documentation. For example here is the curl request that returns 302 to me:

```
url -X GET "https://api.nylas.com/oauth/authorize?client_id=XXXXXXX&redirect_uri=http%3A%2F%2Flvh.me%3A3000%2Fsome_action&response_type=code&scopes=email.drafts%2Cemail.read_only%2Cemail.send%2Cemail.modify" -H "Accept: application/json"
```

and response:

```
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>Redirecting...</title>
<h1>Redirecting...</h1>
<p>You should be redirected automatically to target URL: <a href="https://api.nylas.com/oauth/login?proposal_id=YYYYYYY">https://api.nylas.com/oauth/login?proposal_id=YYYYYYY</a>.  If not click the link.
```

File: spec/nylas/http_client_spec.rb [hunk 1]
```diff
@@ -59,6 +59,16 @@
         url: "https://token:@api.nylas.com/contacts/1234/picture"
       )
     end
+
+    it "handles redirect correctly" do
```
Comment (mrashed-dev) [inline spec/nylas/http_client_spec.rb:63]: Thanks for adding a test case, it's much appreciated! 😄

**Review by mrashed-dev — APPROVED** [review]
Looks good to me! Thanks for contributing! 😃

**Pull request merged**